src/map.cpp

```diff
@@ -8175,22 +8169,25 @@ template<typename Functor>
     }
 }
 
-void map::scent_blockers( bool (&blocks_scent)[SEEX * MAPSIZE][SEEY * MAPSIZE],
-                          bool (&reduces_scent)[SEEX * MAPSIZE][SEEY * MAPSIZE],
+void map::scent_blockers( std::array<std::array<bool, SEEX * MAPSIZE>, SEEY * MAPSIZE> &blocks_scent,
+                          std::array<std::array<bool, SEEX * MAPSIZE>, SEEY * MAPSIZE> &reduces_scent,
```

Comment:
What does this do in the `map` class? The function sees right through the `private` guts of the `scent` class, but the only private thing of `map` that I can see here is `function_over()` (no harm in making it public). This signals me that the function should belong to the `scent` class too.